src/kernel/filesystem/fat32.zig

```diff
@@ -3837,87 +3753,195 @@ test "Fat32FS.openImpl - entry iterator failed 2nd next" {
-        // Long entry 2
-        0x02, 0x6E, 0x00, 0x67, 0x00, 0x76, 0x00, 0x65, 0x00, 0x72, 0x00, 0x0F, 0x00, 0x6E, 0x79, 0x00,
-        0x6C, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x00, 0x00, 0x6F, 0x00, 0x6F, 0x00,
-        // Long entry 1
-        0x01, 0x6C, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x0F, 0x00, 0x6E, 0x6F, 0x00,
-        0x6E, 0x00, 0x67, 0x00, 0x6C, 0x00, 0x6F, 0x00, 0x6F, 0x00, 0x00, 0x00, 0x6F, 0x00, 0x6F, 0x00,
-        // Short entry
-        0x4C, 0x4F, 0x4F, 0x4F, 0x4F, 0x4F, 0x7E, 0x31, 0x54, 0x58, 0x54, 0x00, 0x18, 0xA0, 0x68, 0xA9,
-        0xFE, 0x50, 0x00, 0x00, 0x00, 0x00, 0x6E, 0xA9, 0xFE, 0x50, 0x08, 0x00, 0x13, 0x00, 0x00, 0x00,
-    };
+    vfs.setRoot(test_fs.root_node.node);
 
-    // Goto root dir and write a long and short entry
-    const sector = test_fs.fat_config.clusterToSector(test_fs.root_node.cluster);
-    try test_fs.stream.seekableStream().seekTo(sector * test_fs.fat_config.bytes_per_sector);
-    try test_fs.stream.writer().writeAll(entry_buff[0..]);
+    const test_node = try vfs.openFile("/short.txt", .NO_CREATION);
+    defer test_node.close();
 
-    const file_node = try Fat32FS(@TypeOf(stream)).openImpl(test_fs.fs, &test_fs.root_node.node.Dir, "looooongloooongveryloooooongname.txt");
-    defer file_node.File.close();
+    var buff = [_]u8{0xAA} ** 16;
+    const read = try test_node.read(buff[0..]);
+    expectEqualSlices(u8, buff[0..read], "short.txt");
+    // The rest should be unchanged
+    expectEqualSlices(u8, buff[read..], &[_]u8{0xAA} ** 7);
 }
 
-test "Fat32FS.openImpl - no match" {
-    var stream = &std.io.fixedBufferStream(test_stream_buff[0..]);
+test "Fat32FS.read - large" {
+    const test_fat32_image = try std.fs.cwd().openFile("test/fat32/test_fat32.img", .{});
+    defer test_fat32_image.close();
 
-    try mkfat32.Fat32.make(.{}, stream, true);
+    var test_fs = try initialiseFAT32(std.testing.allocator, test_fat32_image);
+    defer test_fs.destroy();
 
-    var test_fs = try initialiseFAT32(std.testing.allocator, stream);
+    vfs.setRoot(test_fs.root_node.node);
+
+    const test_node = try vfs.openFile("/large_file2.txt", .NO_CREATION);
+    defer test_node.close();
+
+    var buff = [_]u8{0xAA} ** 8450;
+    const read = try test_node.read(buff[0..]);
+    expectEqual(read, 8450);
+    // The buffer should have:
+    // A ** 64\n
+    // BBB..
+    // CCC..
+    // .....
+    // ZZZ..
+    // 5 times
+    var i: usize = 0;
+    // Repeats 5 times
+    while (i < 5) : (i += 1) {
+        // 26 characters
+        var j: usize = 0;
+        while (j < 26) : (j += 1) {
+            // Each character repeated 64 times
+            var k: usize = 0;
+            while (k < 64) : (k += 1) {
+                expectEqual(buff[(i * 65 * 26) + (j * 65) + k], 'A' + @intCast(u8, j));
+            }
+            // New line
+            expectEqual(buff[(i * 65 * 26) + (j * 65) + 64], '\n');
+        }
+    }
```

Comment:
Instead of this, you could use `@embedFile("test/fat32/test_files/large_file.txt")` to get a slice of the file's contents and do a slice equality check. It's up to you depending on if you think it's cleaner or not!

Reply:
That is a good idea